### PR TITLE
Add Ethiopian date widget to webapps

### PR DESCRIPTION
**Pull request by proteusvacuum**
## Summary

This adds the ethiopian date picker from https://github.com/kbwood/calendars for use in WebApps. 
We will be adding other translations for this widget directly in the codebase once we have those from the partner, but wanted to get this out for review and QA. 

https://docs.google.com/document/d/1ZCBQxZOIxUB56NgYg6fKVoX-n_DctsR0VG9fvtRtWI8/edit#
https://dimagi-dev.atlassian.net/browse/SC-1127

## Product Description

Allows users to use the `ethiopian` calendar in WebApps, as they do on [Android](https://confluence.dimagi.com/display/commcarepublic/Alternate+Calendar+Formats+for+CommCare+Android)

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I am certain that this PR will not introduce a regression for the reasons below

### Automated test coverage

This adds a test to the cloudcare test suite that tests changing and updating the widget. However, since this is mostly a UI feature, manual testing will be done via the QA team on staging. 

### Safety story

I have done a lot of local testing in WebApps and App Preview. This is going to go through QA. This widget is only activated in very limited circumstances (when the appearance attribute of a date question is set to "ethiopian")

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/entrycontrols_full_spec.js [hunk 1]
```diff
@@ -274,6 +274,35 @@ describe('Entries', function () {
...
+        questionJSON.datatype = Const.DATE;
+        questionJSON.answer = '2021-01-29'; // 2013-05-21 in Ethiopian
+        questionJSON.style = { raw: 'ethiopian' };
+
+        var entry = UI.Question(questionJSON).entry;
+        entry.entryId = 'fake-div';
+        assert.isTrue(entry instanceof Controls.EthiopianDateEntry);
+        assert.equal(entry.templateType, 'ethiopian-date');
+
+        $(document.body).append("<div id='"+ entry.entryId + "'></div>");
```
Comment (proteusvacuum) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/entrycontrols_full_spec.js:287]: I didn't see any other tests adding items to the DOM elsewhere, but I wanted to be able to test interaction with the widget. Would love feedback on whether this is a bad idea.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js [hunk 2]
```diff
@@ -1030,7 +1076,11 @@ hqDefine("cloudcare/js/form_entry/entrycontrols_full", function () {
                 entry = new MultiSelectEntry(question, {});
                 break;
             case Const.DATE:
-                entry = new DateEntry(question, {});
+                if (style === 'ethiopian') {
```
Comment (orangejenny) [inline corehq/apps/cloudcare/static/cloudcare/js/form_entry/entrycontrols_full.js:1079]: Most (all?) of the other styles are added as constants in const.js.

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/entrycontrols_full_spec.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/entrycontrols_full_spec.js]: It's more typical to put DOM for tests in the `fixtures` block in the mocha template, would that work here? I'm curious, what is it that doesn't work if the question isn't actually part of the DOM? Do the `entry.$picker` functions not work?

Comment (proteusvacuum) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/entrycontrols_full_spec.js]: Oh that's what I was looking for, thanks! Yes, it's the `entry.$picker.calendarsPicker` functions that don't work. I think this is in large part due to how this calendar widget is written as a jquery plugin.

Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/spec/entrycontrols_full_spec.js]: cool, I can imagine the plugin designers assuming they're on a real page

**Review by orangejenny — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]

**Review by kaapstorm — APPROVED** [review]

**Pull request merged**